pytorch_blade/src/ltc/disc_compiler/passes/disc_fuser.cpp

```diff
@@ -101,12 +102,22 @@ void DiscFusion(const std::shared_ptr<Graph>& graph) {
   CastingScalarInputToTensor(graph);
 }
 
+torch::Device getInputDevice(torch::jit::Node* node) {
+  for (auto input : node->inputs()) {
+    if (input->type()->cast<c10::TensorType>() != nullptr) {
+      return input->type()->cast<c10::TensorType>()->device().value();
+    }
+  }
+  return torch::Device(torch::kCUDA, c10::cuda::current_device());
```

Comment:
How about raising an exception here? Since that it is error-prone.

Reply:
better to assign a default device, maybe the original sub-graph only include scalar input.

Reply:
I think scalar input should always be on the host side. And we should avoid TensorType without device setting.

Reply:
After offline discussion, assign the scalar input to the CUDA device always.